src/apps/patron-page/sections/PincodeSection.tsx

```diff
@@ -85,7 +89,11 @@ const PincodeSection: FC<PincodeSectionProps> = ({
           type="password"
           onChange={(newPin) => setConfirmPincode(newPin)}
           value={confirmPincode}
-          label={t("patronPageConfirmPincodeLabelText")}
+          label={
+            required
+              ? `${t("patronPageConfirmPincodeLabelText")}*`
+              : t("patronPageConfirmPincodeLabelText")
```

Comment:
I think it should be the responsibility of the `TextInput` component to prefix label text with an asterisk if it is required.